### PR TITLE
feat(metrics): create default metrics server

### PR DESCRIPTION
**Pull request by ethanfrogers**
provide a default metrics server implementation and common http request
instrumentation. this metrics utility can be used in any application
that needs to conform to the patterns we need for observability.

**the implementation of all the requires labels for metrics is incomplete** but i wanted to get something out that could start being used.

### CODE REVIEW

**Comment by ethanfrogers** [comment]
## Example HTTP request metrics

```
# HELP metrics_example_canalsEndpoint_latency metrics_example_canalsEndpoint_latency
# TYPE metrics_example_canalsEndpoint_latency summary
metrics_example_canalsEndpoint_latency{endpoint="canals",quantile="0.5"} 0.0023159999400377274
metrics_example_canalsEndpoint_latency{endpoint="canals",quantile="0.9"} 0.007306999992579222
metrics_example_canalsEndpoint_latency{endpoint="canals",quantile="0.99"} 0.007306999992579222
metrics_example_canalsEndpoint_latency_sum{endpoint="canals"} 0.009622999932616949
metrics_example_canalsEndpoint_latency_count{endpoint="canals"} 2
# HELP metrics_example_http_server_requests metrics_example_http_server_requests
# TYPE metrics_example_http_server_requests counter
metrics_example_http_server_requests{method="GET",outcome="SUCCESS",status="200",uri="/canals"} 2
```

File: metrics.go [hunk 1]
```diff
@@ -79,6 +90,19 @@ func (mc *MetricsServer) WatchForShutdown() {
 }
 
 func (ms *MetricsServer) Start() error {
```
Comment (aelath) [inline metrics.go:94]: I noticed that here you use `ms`, but in other methods on type `*MetricsServer` you use `mc`. Does it make sense to to use two different conventions?

**Review by aelath — APPROVED** [review]
LGTM! One comment on consistent naming. No reason to block.

File: metrics.go (hunk 1 above)
Comment (ethanfrogers) [reply metrics.go:94]: No, `mc` is an artifact from when I briefly called it `MetricsCollector` in an attempt to obfuscate the fact that it was a server at all.

**Pull request merged**